cmd/lifecycle/analyzer.go

```diff
@@ -3,139 +3,75 @@ package main
 import (
 	"fmt"
 
-	"github.com/buildpacks/imgutil"
-	"github.com/buildpacks/imgutil/local"
-	"github.com/buildpacks/imgutil/remote"
 	"github.com/docker/docker/client"
 	"github.com/google/go-containerregistry/pkg/authn"
-	"github.com/google/go-containerregistry/pkg/name"
 	"github.com/pkg/errors"
 
-	"github.com/buildpacks/lifecycle"
 	"github.com/buildpacks/lifecycle/auth"
-	"github.com/buildpacks/lifecycle/buildpack"
-	"github.com/buildpacks/lifecycle/cache"
 	"github.com/buildpacks/lifecycle/cmd"
-	"github.com/buildpacks/lifecycle/image"
+	newplat "github.com/buildpacks/lifecycle/cmd/lifecycle/platform"
 	"github.com/buildpacks/lifecycle/internal/encoding"
-	"github.com/buildpacks/lifecycle/internal/layer"
 	"github.com/buildpacks/lifecycle/platform"
 	"github.com/buildpacks/lifecycle/priv"
 )
 
 type analyzeCmd struct {
-	analyzeArgs
-	additionalTags  cmd.StringSlice
-	analyzedPath    string
-	cacheImageRef   string
-	legacyCacheDir  string
-	legacyGroupPath string
-	outputImageRef  string
-	stackPath       string
-	uid, gid        int
-}
-
-// analyzeArgs contains inputs needed when run by creator.
-type analyzeArgs struct {
-	launchCacheDir   string
-	layersDir        string
-	previousImageRef string
-	runImageRef      string
-	skipLayers       bool
-	useDaemon        bool
+	platform Platform
 
-	docker      client.CommonAPIClient // construct if necessary before dropping privileges
-	keychain    authn.Keychain         // construct if necessary before dropping privileges
-	legacyCache lifecycle.Cache
-	legacyGroup buildpack.Group
-	platform    Platform
+	newplat.AnalyzeInputs
+	docker   client.CommonAPIClient // construct if necessary before dropping privileges
+	keychain authn.Keychain         // construct if necessary before dropping privileges
 }
 
 // DefineFlags defines the flags that are considered valid and reads their values (if provided).
 func (a *analyzeCmd) DefineFlags() {
-	cmd.FlagAnalyzedPath(&a.analyzedPath)
-	cmd.FlagCacheImage(&a.cacheImageRef)
-	cmd.FlagGID(&a.gid)
-	cmd.FlagLayersDir(&a.layersDir)
-	cmd.FlagUID(&a.uid)
-	cmd.FlagUseDaemon(&a.useDaemon)
-	if a.platform.API().AtLeast("0.9") {
-		cmd.FlagLaunchCacheDir(&a.launchCacheDir)
-		cmd.FlagSkipLayers(&a.skipLayers)
-	}
-	if a.platformAPIVersionGreaterThan06() {
-		cmd.FlagPreviousImage(&a.previousImageRef)
-		cmd.FlagRunImage(&a.runImageRef)
-		cmd.FlagStackPath(&a.stackPath)
-		cmd.FlagTags(&a.additionalTags)
-	} else {
-		cmd.FlagCacheDir(&a.legacyCacheDir)
-		cmd.FlagGroupPath(&a.legacyGroupPath)
-		cmd.FlagSkipLayers(&a.skipLayers)
+	switch {
```

Comment:
I love the switch, keeping track of what was being registered was super annoying before.